app/controllers/jobseekers/passwords_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Jobseekers::PasswordsController < Devise::PasswordsController
+  before_action :set_jobseeker_from_token, only: %i[edit update]
+  before_action :check_reset_password_token, only: %i[edit]
+  after_action :remove_devise_notice_flash!, only: %i[create]
+
+protected
+
+  def reset_password_token
+    Devise.token_generator.digest(resource_class, :reset_password_token, params[:reset_password_token] || params[resource_name][:reset_password_token])
+  end
+
+  def set_jobseeker_from_token
```

Comment:
I'm surprised we don't get more of this for free

Reply:
the problem is that at that stage it does not retrieve the model by default. Only when you submit the token and the new password.